sesman/env.c

```diff
@@ -200,3 +201,45 @@ env_set_user(const char *username, char **passwd_file, int display,
 
     return error;
 }
+
+
+/******************************************************************************/
+int DEFAULT_CC
+env_add_xauth_user(int display, char *cookie, char *file)
+{
+    FILE *dp;
+    char xauth_str[256];
+    int fd;
+
+    if ( file == NULL )
+    {
+        if (!g_file_exist(".Xauthority")) 
+        {
+            fd = g_file_open(".Xauthority");
+            g_file_close(fd);
+        }
+
+        g_sprintf(xauth_str, "xauth -q add :%d . %s", display, cookie);
+    }
+    else
+    {
+        if (!g_file_exist(file))
+        {
+            fd = g_file_open(file);
+            g_file_close(fd);
+        }
+
+        g_sprintf(xauth_str, "xauth -q -f %s  add :%d . %s", 
+                file, display, cookie);
+    }
+    log_message(LOG_LEVEL_DEBUG, "xauth command: %s", xauth_str);
+
+    if ((dp = popen(xauth_str, "r")) == NULL) {
+        log_message(LOG_LEVEL_INFO, "xauth failed, no X security");
```

Comment:
No way it's LOG_LEVEL_INFO. I would use LOG_LEVEL_ERROR; the function even returns an error code. Also, it's not xauth that failed here. It could not be run. Failures of xauth would be in pclose() return code.